talk/basicconcepts/control.tex

```diff
@@ -221,6 +221,30 @@
   \end{exampleblock}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitlecpp[20]{init-statements for range based loops}
+  \begin{block}{}
+    Allows to limit variable scope in range based loops
+  \end{block}
+  \begin{alertblock}{\cpp17}
+    \begin{cppcode*}{}
+      std::array data = {"hello", ",", "world"};
+      std::size_t i = 0;
+      for (auto& d : data) {
+        std::cout << i++ << ' ' << d << '\n';
+      }
+    \end{cppcode*}
+  \end{alertblock}
+  \begin{exampleblock}{\cpp20}
+    \begin{cppcode*}{}
+      std::array data = {"hello", ",", "world"};
+      for (std::size_t i = 0; i < data.size(); ++i) {
+        std::cout << i << ' ' << data[i] << '\n';
+      }
```

Comment:
So, what does this example show? There is no range-based for loop :)

Reply:
damn, I copy pasted the wrong lines. Should have been obviously : 
```cpp
      for (std::size_t i = 0; auto& d : data) {
        std::cout << i++ << ' ' << d << '\n';
      }
```